Correct SPDX license identifiers

keithc-ca opened the pull request:
Issue #6995:
* "GPL-2.0" should have been "GPL-2.0-only"
* "OpenJDK-assembly-exception-1.0" replaces "Assembly-exception"

mstoodle reviewed (changes requested):
haven't checked everything yet, but the mentioned missing characters cause build breaks

keithc-ca commented on jitbuilder/apigen/schema/api.schema.json:
```diff
@@ -4,7 +4,7 @@
     "description": "Format for describing framework and library APIs in a way that allows simple generation of language bindings",
     "$comment": "Copyright IBM Corp. and others 2018 (see LICENSE.md)",
-    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception",
+    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
```
Corrected in 62c1cd97089e589f0a2b440fd70929090753f153.

mstoodle commented on jitbuilder/apigen/schema/api-class.schema.json:
```diff
@@ -4,7 +4,7 @@
     "description": "Format for specifying API classes",
     "$comment": "Copyright IBM Corp. and others 2018 (see LICENSE.md)",
-    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception",
+    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
```
same missing ", at end of line

mstoodle commented on jitbuilder/apigen/schema/api-field.schema.json:
```diff
@@ -4,7 +4,7 @@
     "description": "Format for specifying API fields",
     "$comment": "Copyright IBM Corp. and others 2018 (see LICENSE.md)",
-    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception",
+    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
```
missing ",

mstoodle commented on jitbuilder/apigen/schema/api-service.schema.json:
```diff
@@ -4,7 +4,7 @@
     "description": "Format for specifying API services",
     "$comment": "Copyright IBM Corp. and others 2018 (see LICENSE.md)",
-    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception",
+    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
```
missing ",

mstoodle commented on jitbuilder/apigen/schema/api-type.schema.json:
```diff
@@ -4,7 +4,7 @@
     "description": "Format for specifying API datatypes",
     "$comment": "Copyright IBM Corp. and others 2018 (see LICENSE.md)",
-    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception",
+    "$comment": "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
```
missing ",

mstoodle reviewed (changes requested):
even json file has the same issue of missing ", at the end of the line that's been updated (except the first one I pointed out which has been fixed - thanks!)

keithc-ca commented:
Hopefully I've fixed all files in 0972f4a4e2f7436e1dd51bea8156979175804c81.

keithc-ca commented:
Besides the json files, there were others that didn't get (mechanically) updated properly. Marking as draft until those are corrected.

keithc-ca commented:
Rebased and updated:
* fix new files added recently
* fix `*.s` files where long lines are not permitted
* correct files where the license identifier expression spanned two or more lines
* fix strange "SPDX-L$(LIBJITBUILDERDIR)cense-Identifier"

keithc-ca commented:
Removed "fixes" comment; new files added by recent pull requests will need to be updated.

mstoodle reviewed (approved):
I found `doc/ValgrindMemcheckAPI.md` which does not have the right license string (only "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0" which is, I think, our original identifier so has been missed for a long time :( ). We can handle it in this PR or under a separate one since it's a long standing mistake.

The failure on MacOS is not due to this change (another instance of #6516)

mstoodle commented:
jenkins build all

keithc-ca commented:
I left `doc/ValgrindMemcheckAPI.md` as it was. It didn't have any invalid identifiers discussed in the issue. If I recall correctly, there were other such files. If the license in those files should be updated, I think that should be handled separately.

AdamBrousseau commented:
jenkins buld riscv

AdamBrousseau commented:
jenkins build riscv

keithc-ca commented:
Is this waiting for additional committer approvals, or something else?
